### PR TITLE
orcid: addition of ORCID module 

### PR DESCRIPTION
**Pull request by drjova**

### CODE REVIEW

**Comment by drjova** [comment]
@michamos all the tests should pass now 🌮 🌮

File: tests/integration/orcid/fixtures/test_orcid_domain_models_TestOrcidPusherDuplicatedIdentifier_clashing.json [hunk 1]
```diff
@@ -31,5 +31,6 @@
   "collaborations": [],
   "document_type": ["article"],
   "preprint_date": "2017-05",
-  "control_number": 666
+  "control_number": 666,
+  "deleted": true
```
Comment (michamos) [inline tests/integration/orcid/fixtures/test_orcid_domain_models_TestOrcidPusherDuplicatedIdentifier_clashing.json]: I'm not sure this is correct, as you're not pushing that record if it's deleted from the start. If you look at the cassette, you'll see that you're just pushing the other record and that's it, so you're not testing this feature.

**Comment by drjova** [comment]
@michamos all green! could you please have a look? I'll squash them before merging

File: tests/integration/orcid/test_orcid_domain_models.py [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+            "control_number"
+        ]
+
+        self.inspire_record = factory.inspire_record
+        self.clashing_record = self.factory_clashing.inspire_record
+        # Disable logging.
+        logging.getLogger("inspirehep.orcid.domain_models").disabled = logging.CRITICAL
+        # self.putcode = ""
+        # self.cache.write_work_putcode(self.putcode)
+        # self.cache2 = OrcidCache(self.orcid, "666")
```
Comment (michamos) [inline tests/integration/orcid/test_orcid_domain_models.py]: leftovers?

File: tests/integration/orcid/test_orcid_domain_models.py [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+
+    def teardown(self):
+        self.cache.delete_work_putcode()
+        # self.cache2.delete_work_putcode()
```
Comment (michamos) [inline tests/integration/orcid/test_orcid_domain_models.py]: same

File: tests/integration/orcid/cassettes/TestOrcidPusherDuplicatedIdentifier.test_happy_flow.yaml [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+      - keep-alive
+      Content-Length:
+      - '0'
+      Content-Type:
+      - application/orcid+json
+    method: DELETE
+    uri: https://api.orcid.org/v2.0/0000-0003-1134-6827/work/57442758
+  response:
+    body:
+      string: '{"response-code":404,"developer-message":"404 Not Found: The resource
```
Comment (michamos) [inline tests/integration/orcid/cassettes/TestOrcidPusherDuplicatedIdentifier.test_happy_flow.yaml]: This seems wrong. As I said in a previous comment, you don't seem to be testing the case where you have duplicate identifiers, as the other paper is not there on ORCID at the beginning.

**Pull request merged**